v3/lints/rfc/lint_dnsname_hyphen_in_sld.go

```diff
@@ -0,0 +1,58 @@
+package rfc
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type DNSNameHyphenInSLD struct{}
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_rfc_dnsname_hyphen_in_sld",
+		Description:   "DNSName should not have a hyphen beginning or ending the SLD",
```

Comment:
To confirm - the reason for limiting this to the second-level domain is the assumption that domain owners don't pay attention to the relevant RFCs and stick all sorts of stuff here (e.g. underscores, leading/trailing -), and since name resolution libraries don't fast-fail, ZLint doesn't bother? :)

(A very valid reason, but just confirming if that was the case)